### ISSUE TITLE
Experiment Module: Improve Visualisation of Related Entities

### ISSUE THREAD

**Issue by aislinnpearson** [Rothamsted-Ecoinformatics/farm_rothamsted#524]
One of the major pieces of feedback that we have had on the experiment module is that it is tricky to navigate. The most common problem is that people create plans rather than proposals. I discussed this with Jonah and we came up with the following concepts which  need further discussion with FFEC/ Mike and Paul:

1. On the proposal entity, once a proposal is approved it would be nice for the experiment, design and plan columns to appear at that stage - perhaps with a click button that says 'Add' if the field is blank (see my terrible drawing below).
2. It would be nice to be able to click on a proposal and see a **filtered** version of all the research programs, experiments, designs, plans and researchers associated with it (see terrible drawing 2 below). I think it is worth considering adding this same view to all the entities.
3. For the scientists, we might want to introduce a new view and/or simplify the Research view so they can **only see their own proposals** and track their progress through the system. This would be need to be linked to as simple way to create researchers, research programs and proposals.

We might want to make additional changes after this to further improve the visualisation/ user experience

![image](https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/assets/16349781/5f762e91-7817-4c08-aa97-c44f1bcc3f10)

![image](https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/assets/16349781/70768306-1b00-483d-bc62-7b1656291479)

**Comment by paul121** [comment]
@aislinnpearson I've started a concept of a "Related" tab that we could use to visualize this information. We could name the tab "Related" or even something like "Research Structure". Right now I have this opening a modal (similar to popup) that clearly lists related Programs, Experiments, Designs, Plans and Proposals.

I think this is a good solution for these use-cases because there will often only be a few related items, not so many that a fitlerable table would be useful. In the cases that there are many items this proposed UI would not be great - but we could make a list item "10 plans" -> that links to a filtered table of the relevant research entity.

Right now these screenshots are display the entity types in a weird order (reverse alphabetical I think?) but it should probably be 1) Programs 2) Proposals 3) Experiments 4) Designs 5) Plans ? Let's chat tomorrow about this!

![program-tabs](https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/assets/3116995/dd5c5259-7bcc-4c91-a071-1fd8bb14b43a)
![program-related](https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/assets/3116995/64d42f64-009d-4fe2-91f4-146fe2971078)
![plan-related](https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/assets/3116995/91e6a08f-65bd-49a0-9c0b-0e555896c7d9)

**Comment by aislinnpearson** [comment]
Hi @paul121 

A couple thoughts:

> I've started a concept of a "Related" tab that we could use to visualize this information. We could name the tab "Related" or even something like "Research Structure".

yes. We like this visualisation! :) I'd leave it as 'related' - it's simple and makes sense

> I think this is a good solution for these use-cases because there will often only be a few related items, not so many that a filterable table would be useful. In the cases that there are many items this proposed UI would not be great - but we could make a list item "10 plans" -> that links to a filtered table of the relevant research entity.

I'm not sure what will happen in the longer term when the system starts being used more extensively... experiments like Broadbalk will build up a lot of plans but only have one design, while other annual experiments that have been going for 20 years might have 20 plans and 20 designs... Saying that, 10 years is a long time, and it's a bit difficult to build a system with that time frame in mind... We are however in the process of retrospectively adding one of our experiments, so maybe once we close off this issue I mark it as for monitoring and see how those look/ if that informs any additional changes we want to make?

> Right now these screenshots are display the entity types in a weird order (reverse alphabetical I think?) but it should probably be 1) Programs 2) Proposals 3) Experiments 4) Designs 5) Plans? 

Agreed!

A couple other points:

1. Can we do this for plots as well? I noticed the other day that there is no easy way to navigate back to a plan or design from an individual plot. Perhaps in this instance we might also want to quick link to the full list of plots and variables, as well as all the other aspects? Maybe something to chat about...

2. A minor thing - I imagine we would get rid of the bits that say 'experiment' etc so it just says related?
![image](https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/assets/16349781/c17d950c-44b9-48ee-a7b5-90b543308404)

**Comment by aislinnpearson** [comment]
Aislinn to make a new issue for the plots. Paul to let Aislinn to know when complete so can be moved to a monitor issue

**Comment by aislinnpearson** [comment]
This looks so good @paul121. I think at some point in the future we might want to look at how we organise things - e.g. proposals with all the links underneath or plans, but let's test it as it is first. Do you have plans for how this will work with thigns like Plant Assets? I see it appears on all entities (which again I think is great)

**Comment by paul121** [comment]
> I see it appears on all entities (which again I think is great)

To start, I will make sure this is hidden on all entities unless there is a relationship!

In the future, yes, we can expand how this works and how it is organised. I think this could be abstracted for use in farmOS more generally.

**Comment by paul121** [comment]
> To start, I will make sure this is hidden on all entities unless there is a relationship!

Fixed
